lib/libbento4/Core/Ap4CommonEncryption.h

```diff
@@ -523,7 +523,7 @@ class AP4_CencSingleSampleDecrypter
     virtual AP4_UI32 AddPool() { return 0; };
     virtual void RemovePool(AP4_UI32 poolid) {};
     virtual const char* GetSessionId() { return NULL; };
-
+    virtual const char* GetChallengeB64Data() { return NULL; };
```

Comment:
Is it really good practice to directly patch a third party library instead of PR-ing the change upstream and to use a patch file here until the upstream PR got merged and libbento used here was updated?

Reply:
Good point however unfortunately we're already sitting on top of several years of direct patches (lots) on a very outdated libbento, it really needs to be brought up to date. I tried last year sometime but it introduced bugs :/

To address this issue @CastagnaIT you should be able to accomplish what you want without changes to libbento - introduce the new function in SSD_dll.h

Reply:
> introduce the new function in SSD_dll.h

I'm wrong about this (was thinking of `decryptor_` not `single_sample_decrypter_`), changes need to stay in libbento for now.